src/background/waveform_backend.js

```diff
@@ -0,0 +1,62 @@
+import Logger from "../logger";
+
+export default class WaveformBackend {
+  constructor() {
+    this.log = new Logger();
+    this.boundProcessAudio = WaveformBackend.processAudio.bind(this);
+  }
+
+  init() {
+    this.log.info("starting waveform backend.");
+    chrome.runtime.onMessage.addListener(this.boundProcessAudio);
+  }
+
+  static processAudio(request, sender, sendResponse) {
+    if (!(request.contentScriptQuery == "renderBuffer")) return;
```

Comment:
@emcniece yeah, this is a common pattern in C++, instead of having all the code in an `if` block, you negate the logic and return. This way, the code is much cleaner and it is in general very easy to read.

Reply:
@emcniece oh man, yes, so silly